client/proxy/proxy.go

```diff
@@ -21,6 +21,15 @@ type ManagerClient struct {
 	client           *http.Client
 }
 
+// Client defines the interface for the proxy specific functionality of the manager client
+type Client interface {
+	GetConfig(model.Key) (*apiserver.Config, error)
```

Comment:
Ah crap, this was all fine as I parsed it client side but then I removed it so we could reduce the client side validation...